client/src/app/site/motions/services/diff.service.ts

```diff
@@ -2006,4 +2053,50 @@ export class DiffService {
 
         return html;
     }
+
+    /**
+     * This is used to extract affected lines of a paragraph with the possibility to show the context (lines before
+     * and after) the changed lines and displaying the line numbers.
+     *
+     * @param paragraphNo
+     * @param origText - needs to be line-numbered
+     * @param newText
+     * @param lineLength
+     * @return DiffLinesInParagraph
+     */
+    public getAmendmentParagraphsLinesByMode(paragraphNo: number, origText: string, newText: string, lineLength: number): DiffLinesInParagraph {
+        const paragraph_line_range = this.lineNumberingService.getLineNumberRange(origText),
+            diff = this.diff(origText, newText),
+            affected_lines = this.detectAffectedLineRange(diff);
+
+        let textPre = '';
+        let textPost = '';
+        if (affected_lines.from > paragraph_line_range.from) {
+            textPre = this.formatDiffWithLineNumbers(
+                this.extractRangeByLineNumbers(diff, paragraph_line_range.from, affected_lines.from),
+                lineLength, paragraph_line_range.from
+            );
+        }
+        if (paragraph_line_range.to > affected_lines.to) {
+            textPost = this.formatDiffWithLineNumbers(
+                this.extractRangeByLineNumbers(diff, affected_lines.to, paragraph_line_range.to),
+                lineLength, affected_lines.to
+            );
+        }
+        const text = this.formatDiffWithLineNumbers(
+            this.extractRangeByLineNumbers(diff, affected_lines.from, affected_lines.to),
+            lineLength, affected_lines.from
+        );
+
+        return {
```

Comment:
this returns "some" object. I would advise to use the interface in the function or use `return {...} as DiffLinesInParagraph`